cocos/network/WebSocket.h

```diff
@@ -238,7 +239,7 @@ class CC_DLL WebSocket
 
     struct lws*         _wsInstance;
     struct lws_context* _wsContext;
-    std::shared_ptr<bool> _isDestroyed;
+    std::shared_ptr<std::atomic<bool>> _isDestroyed;
```

Comment:
Since we use multi-thread, sendMessageaToCocosThread & sendMessageToWebsocketThread are asynchronous.
the Websocket instance may be destroyed while sending messages, therefore we do something in the callback, it will lead to dangerous instance invocation. 

Fortunately, C++11 supports lambda to catch a variable, in the destructor of WebSocket, we set the `_isDestroyed` to true, if callback is triggered after WebSocket destruction, we could check `isDestroyed` to decide whether to continue.